### PR TITLE
Hydrogen Gas Turbines and Retrofitting of Gas Turbines

### PR DESCRIPTION
**Pull request by toniseibold**
Adding the following configs
```
electricity:
  H2_plants_DE:
    enable: true
    retrofit_CH4_to_H2:
      enable: true
      efficiency_loss: 0.05
      cost_increase: 0.15
      start: 2025
      force: 2035
```

### modify_prenetwork
- if `config["H2_plants_DE"]["enable"] the extendable links for OCGT, CCGT and gas CHPs in the current planning horizon are copied and added so hydrogen turbines can be built as well
- if the retrofitting of existing OCGT, CCGT, gas CHPs in enabled, existing gas plant links are copied and added as retrofitted H2 links (boundary condition is set in additional functionality) from `config["start"]`on
- if the `config["force"] is set, from the force year on, all OCGT, CCGT, gas CHPs links from `config["start"] on are dropped and only the hydrogen links exist

### additional_functionality
- in case that `config["retrofit_CH4_to_H2"]["enable"]` is set, gas plant links and hydrogen retro links are added as a boundary condition in a way, that the installed capacity of both must be lower equal than the former installed capacity of gas plants

Open issues:
- [x] check the efficiency decrease
- [x] check the costs for retrofitting

### CODE REVIEW

File: workflow/scripts/additional_functionality.py [hunk 1]
```diff
@@ -415,6 +415,42 @@ def add_h2_derivate_limit(n, snapshots, investment_year, config):
                 carrier_attribute="",
             )
 
+def retrofit_gas_plants(n, params, year):
+
+    if params["force"] >= year:
+        logger.info("Retrofitting already forced.")
+    
+    if params["start"] <= year:
```
Comment (nworbmot) [inline workflow/scripts/additional_functionality.py:423]: this should be elif?

**Comment by toniseibold** [comment]
Model results for 3H resolution KN2045_Bal_v4 scenario:
image
Open Cycle capacity agrees with the latest [model run](https://tubcloud.tu-berlin.de/apps/files/?dir=/Shared/ensys_intern/Projekte/2023-Ariadne2/run_results/20240712industrysteam&fileid=4046145748) but model is now also favoring H2 gas turbines:
image

File: workflow/scripts/modify_prenetwork.py [hunk 2]
```diff
@@ -679,6 +679,111 @@ def aladin_mobility_demand(n):
...
+
+        h2_plants = n.links.loc[gas_plants].copy()
+        h2_plants.carrier = h2_plants.carrier.str.replace(
+            carrier, "H2 retro " + carrier
+        )
+        h2_plants.index = h2_plants.index.str.replace(carrier, "H2 retro " + carrier)
+        h2_plants.bus0 = h2_plants.bus0.str.replace("gas", "H2")
+        h2_plants.bus2 = ""
+        h2_plants.efficiency -= params["efficiency_loss"]
+        h2_plants.efficiency2 = 1  # default value
```
Comment (lindnemi) [inline workflow/scripts/modify_prenetwork.py:760]: it's a bit strange that the default value is 0 for H2 plants, but 1 for H2 retro plants

File: config/config.yaml [hunk 3]
```diff
@@ -479,6 +479,12 @@ plotting:
     load: "#111100"
     H2 pipeline (Kernnetz): '#6b3161'
     renewable oil: '#c9c9c9'
+    urban central H2 retro CHP: '#ff0000'
```
Comment (lindnemi) [inline config/config.yaml:482]: i would prefer if they are called `H2 retrofit` instead of `H2 retro`

**Review by lindnemi — CHANGES_REQUESTED** [review]
looking very good, just two minor style suggestions then it can be merged

**Comment by lindnemi** [comment]
I just noticed that there are several more variables in the exporter that need to be adjusted, e.g. capacity|hydrogen and secondary energy input. basically all lines which have "H2 Fuel Cell" in master

**Comment by lindnemi** [comment]
i updated the exporter

however, the workflow is still failing because of a bug in `make_summary`. We should figure that out before merging :thinking:

**Comment by lindnemi** [comment]
plot_summary was failing because the H2 plants were connected to a non-existent `DE H2` bus, and the retrotfitted ones to an equally non existent `DE H2 retro` bus. I corrected these to the spatially resolved `DE1 X H2` buses.

I really wonder why the optimization went through at all and did not brake. @nworbmot @toniseibold 

In the 365H resolution no gas or H2 turbines get built. If this stays the same for 3H we will have to force in the Kraftwerkstrategie plants:

Capacity-Additions-Electricity-Gas

Capacity-Additions-Electricity-Hydrogen


The hydrogen capacity is all fuel cells at the moment.


However, we do see gas boilers being built. Maybe those should be retrofittable to H2 as well?
Capacity-Additions-Heat-Gas

**Pull request merged**